MJAVADOC-431 allow maven descriptor to javadoc jar

peterlynch opened the pull request:
Note: ~~maven-archiver dependency version was upgraded to take advantage of MavenProject.clone(), which avoids having the new unit test from creating a maven-archiver directory in the basedir because the MavenProject() constructor used in 2.5 does not copy the build element.~~ reverted

peterlynch committed:
MJAVADOC-431 allow maven descriptor to javadoc jar

Note: maven-archiver dependency version was upgraded to take advantage of MavenProject.clone(), which avoids having the new unit test from creating a maven-archiver directory in the basedir because the MavenProject() constructor used in 2.5 does not copy the build element.

michael-o commented:
Several issues with the PR commented inline.

michael-o commented on maven-javadoc-plugin/pom.xml:
```diff
@@ -148,7 +148,7 @@ under the License.
       <groupId>org.apache.maven</groupId>
       <artifactId>maven-archiver</artifactId>
-      <version>2.5</version>
+      <version>2.6</version>
```
That should be a separate ticket and not just a mere change in this PR.

peterlynch replied:
Do you want a separate jira issue but include this change only in this PR?

Or do you want a new PR/issue with this change only?

Or do you want me to revert this version change ( which will cause a **maven-archiver** directory to be created in the root of the project after the tests run ).

michael-o replied:
Drop this change. If you want to upate maven archiver, create a separate JIRA ticket for that.

michael-o commented on maven-javadoc-plugin/src/main/java/org/apache/maven/plugin/javadoc/JavadocArchiveConfiguration.java:
```diff
@@ -0,0 +1,36 @@
+    {
+        setAddMavenDescriptor( false );
+    }
+}
```
Add an empty trailing line.

michael-o commented on maven-javadoc-plugin/src/test/java/org/apache/maven/plugin/javadoc/JavadocJarTest.java:
```diff
@@ -109,29 +103,62 @@ else if ( Float.compare( javadocVersion, 1.8f ) < 0 )
+            assertTrue("Expected jar to contain " + entry, set.contains(entry));
+        }
     }
 }
```
The entire file has been unproperly reformatted. Please restore the code style enforced by Apache Maven as well as your new tests.

michael-o commented on maven-javadoc-plugin/src/test/java/org/apache/maven/plugin/javadoc/stubs/JavadocJarArchiveConfigProjectStub.java:
```diff
@@ -0,0 +1,90 @@
+        File projectFile =  new File( getBasedir(), "javadocjar-archive-config.xml" );
+        readModel( new File( getBasedir(), "javadocjar-archive-config.xml" ) );
+
+        setFile(projectFile);
```
Format.

michael-o commented on maven-javadoc-plugin/src/test/java/org/apache/maven/plugin/javadoc/stubs/JavadocJarDefaultMavenProjectStub.java:
```diff
@@ -59,7 +59,7 @@ public JavadocJarDefaultMavenProjectStub()
 
         Build build = new Build();
-        build.setFinalName( "javadocjar-default" );
+        build.setFinalName( "javadocjar-archive-config" );
```
If this is default why has this been changed?

peterlynch replied:
oops

michael-o commented on maven-javadoc-plugin/src/test/resources/unit/javadocjar-archive-config/javadocjar/def/App.java:
```diff
@@ -0,0 +1,50 @@
+        System.out.println( str );
+    }
+
+}
```
Trailing line.

michael-o commented:
@peterlynch Please update your PR according to me comments and I will review again.

peterlynch committed:
revert final name back to default

peterlynch committed:
an attempt to conform to Maven code style and add end of lines on files I touched...

...truth is every other Java file I didn't touch also does not conform to Maven code style - which is why I wasn't too picky in the first place.

peterlynch committed:
revert archiver back to 2.5 - warning: this creates a maven-archiver directory containing a pom.properties file inside of the project basedir.

peterlynch commented:
@michael-o how does it look now?

michael-o commented:
@peterlynch Please replay your changes on top of master again instead of pushing changes/commits. Squash into one and I will review and merge.

michael-o commented:
@peterlynch I am still waiting for an updated PR.

peterlynch committed:
add trailing new line

peterlynch commented:
@michael-o replaced by #70
